### DISCUSSION TITLE
Multiple Time Calibrations on microinverters

### DISCUSSION THREAD

**Discussion by darkazar** [tbnobody/OpenDTU#1635] [Q&A]
Why after some updates before (idk how many before) i have in logs "time calibration" on almost every microinverter.
I have configured NTP in DTU, also i have own NTP Server but this is not the case. 
In old versions of openDTU  I did not have time calibrations 

![time1](https://github.com/tbnobody/OpenDTU/assets/114813235/cda9fcb3-37fc-462f-9bc8-514a3058b976)
![time2](https://github.com/tbnobody/OpenDTU/assets/114813235/ebddaeaa-c64f-45b3-94dc-91050f7c747b)
![time3](https://github.com/tbnobody/OpenDTU/assets/114813235/a0eec143-0c33-45d7-8f3f-7b55c3ece461)

**Comment by stefan123t** [comment]
The description "Time calibration" for Event ID 2 was found in the Hoymiles S-Miles Installer application.
We have therefor renamed Event 2 from "DTU command failed" to "Time calibration" here https://github.com/tbnobody/OpenDTU/commit/dc5b5e837f85aa6a111e52e2917a01e1f71a605a in https://github.com/tbnobody/OpenDTU/issues/121#issuecomment-1752124208.

We still do not know for sure when / why this event occurs, but it most likely has something to do with the timing of request / responses between the OpenDTU and your inverter (think of a heartbeat request). 
Some of the requests we send do have a timestamp included and bad reception between DTU and inverter can cause retransmission of the request or part of the message. Maybe in such cases the timestamp has to be updated / validated ?

If you know why this Event 2 "Time calibration" / "DTU command failed" occurs in your case please let us know.

**Reply by Sn0w3y** [reply]
Have the same Issue here - happend after trying to set PowerLimits in some kind of way (don't know exactly and 100% but its a guess)

Also the inverter is not reachable anymore for some long times if this happens (e.g. for 500 sec.)

Would be great if it would be fixed :(

**Comment by Berondal** [comment]
I have same problem since update to new firmware. Some Hoymiles are not recheable for long time (1 hour+) 
In log I found a lot of Time calibration on missing Hoymiles.
I don't have problems with old firmware

**Comment by distanzcheck** [comment]
Which old firmware version does not have the "error"?
